File: services/brandService.ts
import { supabase } from './supabase';

export const brandService = {
    /** Fetch all brands ordered by position. */
    async getBrands(): Promise<string[]> {
        const { data, error } = await supabase
            .from('brands')
            .select('name')
            .order('position', { ascending: true });
        if (error) {
            console.error('Error fetching brands:', error);
            return [];
        }
        return (data ?? []).map((row: { name: string }) => row.name);
    },

    /** Add a new brand (upsert to avoid duplicates). */
    async addBrand(name: string): Promise<void> {
        const { error } = await supabase
            .from('brands')
            .upsert({ name }, { onConflict: 'name' });
        if (error) console.error('Error adding brand:', error);
    }
};

File: la_ganga_flutter/lib/services/brand_service.dart
import '../supabase/config.dart';

class BrandService {
  final _client = SupabaseConfig.client;

  /// Fetch all brands ordered by position.
  Future<List<String>> fetchBrands() async {
    final response = await _client
        .from('brands')
        .select('name')
        .order('position', ascending: true);
    final data = response as List<dynamic>;
    return data.map((e) => e['name'] as String).toList();
  }

  /// Add a new brand if it does not already exist.
  Future<void> addBrand(String name) async {
    // Upsert to avoid duplicates (name is unique)
    await _client.from('brands').upsert({'name': name}, onConflict: 'name');
  }
}
